eras/alonzo/impl/testlib/Test/Cardano/Ledger/Alonzo/Arbitrary.hs

```diff
@@ -205,14 +208,22 @@ instance
       <*> arbitrary
       <*> arbitrary
 
-instance Era era => Arbitrary (AlonzoScript era) where
-  arbitrary = do
-    lang <- arbitrary -- The language is not present in the Script serialization
-    frequency
-      [ (1, pure (alwaysSucceeds lang 1))
-      , (1, pure (alwaysFails lang 1))
-      , (10, TimelockScript <$> arbitrary)
-      ]
+genAlonzoScript ::
+  forall era.
+  ( Era era
+  ) =>
+  [Language] ->
+  Gen (AlonzoScript era)
+genAlonzoScript langs = do
+  lang <- elements langs -- The language is not present in the Script serialization
+  frequency
+    [ (1, pure (alwaysSucceeds lang 1))
+    , (1, pure (alwaysFails lang 1))
+    , (10, TimelockScript <$> arbitrary)
+    ]
+
+instance forall c. Crypto c => Arbitrary (AlonzoScript (AlonzoEra c)) where
+  arbitrary = genAlonzoScript [PlutusV1]
```

Comment:
heads up @lehins , I did something that I now regret. After I found a way to fix the serialization test, I used `--amend` and forced pushed, and now the relatively small changes I made are lost in the biff diff. apologies.

---

I started trying to use the `Twiddler`, but found that I was still going to have to use newtype wrappers and separate generators. So I tried this instead: have the `Arbitrary (AlonzoScript era)` instance specialized for each era, and passed the languages they support. ie, here you see `Arbitrary (AlonzoScript (AlonzoEra c))`, etc, for each era.

Also, clearly these are not great generators for serialization tests, as we suspected.